engine/src/main/java/org/archive/crawler/frontier/WorkQueueFrontier.java

```diff
@@ -600,54 +602,57 @@ protected CrawlURI findEligibleURI() {
                         }
                     }
                     readyQ = getQueueFor(key);
-                    if(readyQ==null) {
-                         // readyQ key wasn't in all queues: unexpected
-                        logger.severe("Key "+ key +
-                            " in readyClassQueues but not allQueues");
+                synchronized (readyQ) {
+                    if (readyQ == null) {
```

Comment:
This branch is unreachable as synchronised will NullPointerException. Should do the null check before synchronised.

Reply:
I think I've resolved this now.